video.cpp

```diff
@@ -3193,6 +3227,30 @@ void video_cmd(char *cmd)
 			printf("video_cmd: unknown command or format.\n");
 		}
 	}
+
+	// Command to load a custom wallpaper at runtime
+	if (!strncmp(cmd, "load_bg ", 8) &&
+			is_menu())
+	{
+		// Find the direct bg path pointer
+		// The 'cmd' param is max 1024,
+		const char* bg = cmd + 8;
+		while (*bg == ' ' && (bg - cmd) < 1024)
+			bg++;
+
+		// Copy the filename part as this will go invalid soon
+		static char fname[1024];
+		sprintf(fname, "%s", bg);
+
+		// We need to do the load async because we're currently
+		// on the input thread - if we don't use offload we will
+		// stall the core's input processing
+		offload_add_work([=]
+		{
+			load_bg_specific(fname);
+			video_menu_bg(user_io_status_get("[3:1]")); // force redraw
```

Comment:
It is not safe to call video_menu_bg in a separate thread. video_menu_bg makes user io spi calls, if two of those happened from different threads at the same time they would collide with each other and have unpredictable results. I haven't looked at what other global state video_menu_bg might interact with.

Reply:
Fair point, however I could not find any other way to force a call of video_menu_bg. The chance of this happening is low but not impossible - would adding a simple guard suffice (e.g. with a bool that is set upon function entry)? I can amend it to guard against this.

Reply:
Alright, as discussed, I'm going to make sure any SPI stuff is moved outside of the offload - change incoming soon. Afterwards, for maintenance sake I think `video_menu_bg` is a good candidate for refactoring, I can spearhead that effort if you agree.